### PR TITLE
Self-contained dependencies

### PR DESCRIPTION
**Pull request by alejandro-isaza**

### CODE REVIEW

File: docs/Contributing.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Keep this in mind when adding to the library:
 
 * Install Xcode
```
Comment (hewigovens) [inline docs/Contributing.md]: Let's mention `Dockerfile` here as well

File: tools/install-dependencies [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+# Download Nlohmann JSON
+export JSON_VERSION=3.5.0
+JSON_DIR="$ROOT/build/json/staging"
+mkdir -p "$JSON_DIR"
+cd "$JSON_DIR"
+if [ ! -f include.zip ]; then
+    curl -fSsOL https://github.com/nlohmann/json/releases/download/v$JSON_VERSION/include.zip
+fi
+unzip -d "$PREFIX" -f include.zip
```
Comment (hewigovens) [inline tools/install-dependencies]: verify file hash?

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +0,0 @@
-[submodule "lib/googletest"]
```
Comment (hewigovens) [inline .gitmodules]: 👏

File: TrustWalletCore.podspec [hunk 4]
```diff
@@ -158,7 +158,7 @@ Pod::Spec.new do |s|
   s.pod_target_xcconfig = {
     'SYSTEM_HEADER_SEARCH_PATHS' => '$(inherited) /usr/local/include'
   }
-  s.prepare_command = 'tools/generate-files && cmake -H. -Bbuild && make -Cbuild protobuf_ext nlohmann_json'
+  s.prepare_command = 'tools/install-dependencies && tools/generate-files'
```
Comment (hewigovens) [inline TrustWalletCore.podspec]: can we run pod spec lint for this change somehow?

File: docs/Contributing.md (hunk 1 above)
Comment (alejandro-isaza) [reply docs/Contributing.md]: It's mentioned 6 lines below :)

File: tools/install-dependencies (hunk 2 above)
Comment (alejandro-isaza) [reply tools/install-dependencies]: sure

File: TrustWalletCore.podspec (hunk 4 above)
Comment (alejandro-isaza) [reply TrustWalletCore.podspec]: spec lint always fails for me

File: docs/Contributing.md (hunk 1 above)
Comment (hewigovens) [reply docs/Contributing.md]: cool, didn't notice that

File: TrustWalletCore.podspec (hunk 4 above)
Comment (hewigovens) [reply TrustWalletCore.podspec]: ok, I will try it in trust-core

File: tools/install-dependencies [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ROOT="$PWD"
+PREFIX="$ROOT/build/local"
+
+# Setup up folders
+export PATH="$PREFIX/bin":$PATH
+export LDFLAGS="-L$PREFIX/lib"
+export LT_SYS_LIBRARY_PATH="$PREFIX/lib"
+export DYLD_LIBRARY_PATH="$PREFIX/lib"
+
+# Download Google Test
```
Comment (hewigovens) [inline tools/install-dependencies]: we can echo this comment

**Pull request merged**